### ISSUE TITLE
Obsolete attribute with interpolated message doesn't produce diagnostics

### ISSUE THREAD

**Issue by Sonozuki** [dotnet/roslyn#66465]
**Version Used**: 
.NET SDK 7.0.102

**Steps to Reproduce**:

```cs
using System;

namespace ObsoleteExample;

public class Program
{
    public static void Main()
    {
        Interpolation0(); // no diagnostic
        Concatenation0(); // diagnostic
    }

    [Obsolete($"Use {nameof(Interpolation1)}")]
    public static void Interpolation0() { }
    public static void Interpolation1() { }

    [Obsolete("Use " + nameof(Concatenation1))]
    public static void Concatenation0() { }
    public static void Concatenation1() { }
}
```

**Expected Behavior**:
An obsolete diagnostic is issued for the call to Interpolation0().

**Actual Behavior**:
No diagnostic is issued for the call to Interpolation0().

**Comment by Youssef1313** [comment]
Fixed in https://github.com/dotnet/roslyn/pull/64612
